docs/src/high_dimension.rst

```diff
@@ -4,3 +4,145 @@
+
+    # generating the data
+    from hidimstat._utils.scenario import multivariate_simulation_spatial
+    X_init, y, beta, epsilon = multivariate_simulation_spatial(
+        n_samples, shape, roi_size, signal_noise_ratio=10., smooth_X=1
+    )
+
+Then we perform inference on this data using the Desparsified Lasso::
+
+    from hidimstat.desparsified_lasso import (
+    desparsified_lasso,
+    desparsified_lasso_pvalue,
+    )
+    beta_hat, sigma_hat, precision_diagonal = desparsified_lasso(X_init, y)
+    _, pval_corr, _, one_minus_pval_corr, cb_min, cb_max = (
+        desparsified_lasso_pvalue(X_init.shape[0], beta_hat, sigma_hat, precision_diagonal)
+    )
+        
+    # compute estimated support (second method)
+    import numpy as np
+    alpha = .05
+    selected_dl = np.logical_or(pval_corr < alpha , one_minus_pval_corr < alpha)
+    print(f'Desparsified Lasso selected {np.sum(selected_dl)} features among {np.sum(beta > 0)} ')
+
+
+
+Feature Grouping and its shortcomings
+-------------------------------------
+
+As discussed earlier, feature grouping is a meaningful solution to deal with such cases: it reduces the number of features to condition on, and generally also decreases the level of correlation between features.
+
+.. seealso::
+
+   * The :ref:`Grouping documentation <grouping>`
+
+
+As hinted in :footcite:t:`meinshausen2009pvalues` an efficient way to deal with such configuration is to take the per-group average of the features: this leads to a *reduced design*. After inference, all the feature in a given group obtain the p-value of the group representative. When the inference engine is Desparsified Lasso, the resulting method is called Clustered Desparsified lasso, or **CluDL**.
+
+Using the same example as previously, we start by defining a clustering method that will perform the grouping. For image data, Ward clustering is a good default model::
+
+    from sklearn.feature_extraction import image
+    from sklearn.cluster import FeatureAgglomeration
+    n_clusters = 200 
+    connectivity = image.grid_to_graph(n_x=shape[0], n_y=shape[1])
+    ward = FeatureAgglomeration(
+        n_clusters=n_clusters, connectivity=connectivity, linkage="ward"
+    )
+
+Equipped with this, we can use CluDL::
+
+    from sklearn.preprocessing import StandardScaler
+    from hidimstat.ensemble_clustered_inference import (
+        clustered_inference,
+	clustered_inference_pvalue,
+    )
+    ward_, beta_hat, theta_hat, omega_diag = clustered_inference(
+        X_init, y, ward, n_clusters, scaler_sampling=StandardScaler()
+    )
+    _, _, pval_corr, _, one_minus_pval_corr = (
+        clustered_inference_pvalue(X_init.shape[0], False, ward_, beta_hat, theta_hat, omega_diag)
+    )
+
+    # compute estimated support
+    selected_cdl = np.logical_or(pval_corr < alpha, one_minus_pval_corr < alpha)
+    print(f'Clustered Desparsified Lasso selected {np.sum(selected_cdl)} features among {np.sum(beta > 0)} ')
```

Comment:
This code is not executed in the rendered doc. Is there a way to run the code snippets?

Reply:
For the user guide, I don't think that it's a good idea to have code.
In my opinion, it's better to have an example associated with it and make reference to it.